src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -263,6 +268,31 @@ public static Itinerary generateItinerary(GraphPath path, boolean showIntermedia
         return itinerary;
     }
 
+    /**
+     * Iterates over the legs and computes the accessibility scores for transit and walk
+     * legs.
+     */
+    private static void addAccessibilityScore(List<Leg> legs, State[][] states) {
+        RoutingRequest request = states[0][0].getOptions();
+        for (int i = 0; i < legs.size(); i++) {
+            Leg currentLeg = legs.get(i);
+            if (currentLeg.isTransitLeg()) {
+                State[] legStates = states[i];
+                Trip trip = legStates[legStates.length - 1].getBackTrip();
+                Leg nextLeg = null;
+                int nextIndex = i + 1;
+                if(nextIndex < legs.size()){
+                    nextLeg = legs.get(nextIndex);
+                }
+                currentLeg.accessibilityScore =
+                        computeAccessibilityScore(trip, currentLeg, nextLeg, states[i][0].getOptions());
```

Comment:
Could this could be simplified as follows?:

```suggestion
                        computeAccessibilityScore(trip, currentLeg, nextLeg, request);
```